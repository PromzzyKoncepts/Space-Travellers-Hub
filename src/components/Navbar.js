import React from 'react';
import { NavLink } from 'react-router-dom';

import logoImg from '../resources/img/planet.png';

function Navbar() {
  return (
    <nav className="navBar">
      <div className="logoContainer">
        <img className="logo" alt="logo" src={logoImg} />
        <p className="logoText">Space Travelers Hub</p>
      </div>
      <ul className="navigationContainer">
        <li className="navigationItem">
          <NavLink
            className="navLink"
            to="/"
            end
            style={({ isActive }) => ({
              color: isActive ? '#00D7C6' : 'none',
              fontWeight: isActive ? '900' : 'regular',
              fontSize: isActive ? '20px' : 'regular',
            })}
          >
            Rockets
          </NavLink>
        </li>
        <li className="navigationItem">
          <NavLink
            className="navLink"
            to="/missions"
            style={({ isActive }) => ({
              color: isActive ? '#00D7C6' : 'none',
              fontWeight: isActive ? '900' : 'regular',
              fontSize: isActive ? '20px' : 'regular',
            })}
          >
            Missions
          </NavLink>
        </li>
        <li className="navigationItem">
          <NavLink
            className="navLink"
            to="/profile"
            style={({ isActive }) => ({
              color: isActive ? '#00D7C6' : 'none',
              fontWeight: isActive ? '900' : 'regular',
              fontSize: isActive ? '20px' : 'regular',
            })}
          >
            My Profile
          </NavLink>
        </li>
      </ul>
    </nav>
  );
}

export default Navbar;
