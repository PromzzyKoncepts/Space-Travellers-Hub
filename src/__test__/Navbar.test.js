import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter as Router } from "react-router-dom";
import store from "../Redux/ConfigureStore";
import Navbar from "../components/Navbar";

it("renders Component correctly", () => {
  const tree = renderer.create(
    <Router store={store}>
      <Navbar />
    </Router>
  );
  expect(tree).toMatchSnapshot();
});
