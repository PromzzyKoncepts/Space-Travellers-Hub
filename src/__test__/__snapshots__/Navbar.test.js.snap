// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Component correctly 1`] = `
<nav
  className="navBar"
>
  <div
    className="logoContainer"
  >
    <img
      alt="logo"
      className="logo"
      src="planet.png"
    />
    <p
      className="logoText"
    >
      Space Travelers Hub
    </p>
  </div>
  <ul
    className="navigationContainer"
  >
    <li
      className="navigationItem"
    >
      <a
        aria-current="page"
        className="navLink active"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "#00D7C6",
            "fontSize": "20px",
            "fontWeight": "900",
          }
        }
      >
        Rockets
      </a>
    </li>
    <li
      className="navigationItem"
    >
      <a
        className="navLink"
        href="/missions"
        onClick={[Function]}
        style={
          Object {
            "color": "none",
            "fontSize": "regular",
            "fontWeight": "regular",
          }
        }
      >
        Missions
      </a>
    </li>
    <li
      className="navigationItem"
    >
      <a
        className="navLink"
        href="/profile"
        onClick={[Function]}
        style={
          Object {
            "color": "none",
            "fontSize": "regular",
            "fontWeight": "regular",
          }
        }
      >
        My Profile
      </a>
    </li>
  </ul>
</nav>
`;
